torch_geometric/nn/dense/hoscpool.py

```diff
@@ -0,0 +1,153 @@
+import torch
+
+EPS = 1e-15
+
+
+def dense_hoscpool(
```

Comment:
Would this be more suitable as a `Transform`?

Reply:
No, I don't think so. It is a core component of a GNN architecture, so it makes sense to use it as a nn. All baseline pooling operators are contained in torch.nn as well. Besides, we compute and return a loss function inside this function, which would not suit as as `Transform`.

Reply:
Sorry your'e totally right, as mentioned below I misunderstood.

I still am not sure why we wouldn't make this a `torch.nn.Module`. No strong feeling on this, and I do see we have some other examples like this in the library.

@rusty1s any thoughts? I don't know the history of the other dense pooling operator.

Reply:
Hi @Padarn, I am coming back to you because I noticed my PR had not been accepted yet... My bad I did not add the optional example but I just did now. So if everything is ok, could you please merge it ? I would like to community to be able to use my method as soon as possible :)